transports/webtransport/src/transport.rs

```diff
@@ -0,0 +1,682 @@
+        {
+            return Err(TransportError::MultiaddrNotSupported(addr));
+        }
+        Ok((socket_addr, peer_id))
+    }
+}
+
+impl Transport for GenTransport {
+    type Output = (PeerId, Connection);
+    type Error = Error;
+    type ListenerUpgrade = Connecting;
+    type Dial = Pending<Result<Self::Output, Self::Error>>;
+
+    fn listen_on(
+        &mut self,
+        id: ListenerId,
+        addr: Multiaddr,
+    ) -> Result<(), TransportError<Self::Error>> {
+        let (socket_addr, _peer_id) = self.remote_multiaddr_to_socketaddr(addr, false)?;
+        let socket = create_socket(socket_addr).map_err(Self::Error::from)?;
+
+        let server_tls_config = self.config.server_tls_config();
+        let quic_transport_config = self.config.get_quic_transport_config();
+
+        let config = ServerConfig::builder()
+            .with_bind_socket(socket.try_clone().unwrap())
+            .with_custom_tls_and_transport(server_tls_config, quic_transport_config)
+            .build();
+
+        let endpoint =
+            wtransport::Endpoint::server(config).map_err(|e| TransportError::Other(e.into()))?;
+        let keypair = &self.config.keypair;
+        let cert_hashes = self.config.cert_hashes();
+        let handshake_timeout = self.config.handshake_timeout;
+
+        tracing::debug!("Listening on {:?}, listenerId {}", &socket, &id);
+
+        let listener = Listener::new(
+            id,
+            socket,
+            endpoint,
+            keypair,
+            cert_hashes,
+            handshake_timeout,
+        )?;
+        self.listeners.push(listener);
+
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
+        }
+
+        Ok(())
+    }
+
+    fn remove_listener(&mut self, id: ListenerId) -> bool {
+        if let Some(listener) = self.listeners.iter_mut().find(|l| l.listener_id == id) {
+            // Close the listener, which will eventually finish its stream.
+            // `SelectAll` removes streams once they are finished.
+            listener.close(Ok(()));
+
+            tracing::debug!("Listener {id} was removed");
+
+            true
+        } else {
+            false
+        }
+    }
+
+    fn dial(
+        &mut self,
+        _addr: Multiaddr,
+        _opts: DialOpts,
+    ) -> Result<Self::Dial, TransportError<Self::Error>> {
+        Err(TransportError::Other(Error::DialOperationIsNotAllowed))
+    }
+
+    fn poll(
+        mut self: Pin<&mut Self>,
+        cx: &mut Context<'_>,
+    ) -> Poll<TransportEvent<Self::ListenerUpgrade, Self::Error>> {
+        if let Poll::Ready(Some(ev)) = self.listeners.poll_next_unpin(cx) {
+            return Poll::Ready(ev);
+        }
+
+        match &self.waker {
+            None => self.waker = Some(cx.waker().clone()),
+            Some(waker) => {
+                if !waker.will_wake(cx.waker()) {
+                    self.waker = Some(cx.waker().clone());
+                }
+            }
+        };
```

Comment:
Why not just:
```suggestion
        self.waker = Some(cx.waker().clone());
```
I'd prioritize the simpler code over the optimization of avoiding unnecessary waker clones.

Reply:
Wasnt it already concluded that the optimization was not necessary? Either way I do agree that the code being simpler could be better than the optimization version (at least until there is time to show it makes an impact and at that point we could probably work on something to simplify this in general)